plugins/vpp/ifplugin/vppcalls/vpp2001/dump_interface_vppcalls.go

```diff
@@ -296,7 +296,7 @@ func (h *InterfaceVppHandler) DumpMemifSocketDetails() (map[string]uint32, error
 			return memifSocketMap, fmt.Errorf("failed to dump memif socket filename details: %v", err)
 		}
 
-		filename := string(bytes.SplitN(socketDetails.SocketFilename, []byte{0x00}, 2)[0])
+		filename := strings.TrimRight(socketDetails.SocketFilename, "\x00")
```

Comment:
Using `Split` instead of `TrimRight` had a reason, as it happened that the string data (byte array) sometimes contained non zero bytes after a zero byte. In that case the Trimming does not properly return string before the first zero byte. See this example to understand why: https://gist.github.com/ondrej-fabry/0643ea11f81646b189dbc6da2349ae65